src/stores/sdkStore.ts

```diff
@@ -36,163 +30,143 @@ export enum Status {
-  const status = ref(Status.UNINITIALIZED);
-  const address = ref();
-  const networkId = ref();
-  const nodeUrl = ref();
-  const isStatic = ref(true);
+    set(sk: Encoded.AccountSecretKey) {
+      persistSecretKey(sk);
+    },
+  });
 
-  async function scanForWallets() {
-    status.value = Status.WALLET_SCANNING;
+  const aeSdk = new AeSdk({
+    onCompiler: new CompilerHttp(COMPILER_URL),
+    accounts: [new MemoryAccount(secretKey.value)],
+    nodes,
+  });
+  let walletConnector: WalletConnectorFrame | undefined;
 
-    return new Promise<void>((resolve, reject) => {
-      if (aeSdk.value instanceof AeSdkAepp) {
-        const handleWallets: ({
-          wallets,
-          newWallet,
-        }: {
-          wallets: Wallets;
-          newWallet?: Wallet | undefined;
-        }) => Promise<void> = async ({ wallets, newWallet }) => {
-          newWallet = newWallet || Object.values(wallets)[0];
-          stopScan();
-
-          try {
-            (aeSdk.value as AeSdkAepp).disconnectWallet();
-          } catch (_) {
-            // specifically ignore, to ensure we are disconnected before trying to connect, but disconnect will error if we are not connected
-          }
-
-          const { networkId } = await (
-            aeSdk.value as AeSdkAepp
-          ).connectToWallet(newWallet.getConnection());
-
-          (aeSdk.value as AeSdkAepp).selectNode(networkId);
-
-          await (aeSdk.value as AeSdkAepp).subscribeAddress(
-            SUBSCRIPTION_TYPES.subscribe,
-            "current",
-          );
-
-          resolve();
-        };
-        const scannerConnection = new BrowserWindowMessageConnection();
-        const stopScan = walletDetector(scannerConnection, handleWallets);
-      } else reject("AeSdk instance needed for wallet connection");
-    });
+  const status = ref(Status.UNINITIALIZED);
+  const address = ref(aeSdk.address);
+  const nodeUrl = ref(aeSdk.api.$host);
+  const isLocalAccount = ref(true);
+
+  async function disconnectWallet() {
+    if (walletConnector == null) return;
+    walletConnector.disconnect();
+    walletConnector.removeAllListeners();
+    walletConnector = undefined;
   }
 
-  async function initAeSdkAepp(customNodeUrl?: string) {
-    isStatic.value = false;
-    aeSdk.value = new AeSdkAepp({
-      name: "Contract Editor",
-      ...getNodes(customNodeUrl),
-      onCompiler: new CompilerHttp(COMPILER_URL, { ignoreVersion: true }),
-      onNetworkChange: async ({ networkId }) => {
-        if (aeSdk.value) {
-          const [{ name }] = (await aeSdk.value.getNodesInPool()).filter(
-            (node) => node.nodeNetworkId === networkId,
-          );
-          aeSdk.value.selectNode(name);
-          await updateConnectionInfo();
-        }
-      },
-      onAddressChange: () => updateConnectionInfo(),
-      onDisconnect: () => alert("Aepp is disconnected"),
+  async function setupWalletConnector() {
+    status.value = Status.WALLET_SCANNING;
+    disconnectWallet();
```

Comment:
maybe add await or void for this async function call

Reply:
It is a mistake, there is no disconnect confirmation on the wallet side, so `disconnectWallet` runs in sync. I'm removing `async` from its definition.